Glyssen/Rules/CharacterGroupGenerator.cs

```diff
@@ -914,13 +1021,118 @@ public static void DistributeBooksAmongNarratorGroups(List<AuthorStats> authorSt
+			{
+				var narratorPrefix = CharacterVerseData.GetCharacterPrefix(CharacterVerseData.StandardCharacter.Narrator);
+
+				// get standard characters not yet assigned
+				foreach (var character in includedCharacterDetails.Where(c => !Groups.Any(g => g.CharacterIds.Contains(c.CharacterId))).ToList())
+				{
+					var characterType = character.StandardCharacterType;
+					switch (characterType)
+					{
+						case CharacterVerseData.StandardCharacter.BookOrChapter:
+							AddExtraBiblicalToGroupOrNarratorGroup(character.CharacterId, BookTitleChapterGroup, characterType,
+								narratorPrefix, dramatizationPreferences);
+							break;
+
+						case CharacterVerseData.StandardCharacter.ExtraBiblical:
+							AddExtraBiblicalToGroupOrNarratorGroup(character.CharacterId, SectionHeadGroup, characterType,
+								narratorPrefix, dramatizationPreferences);
+							break;
+
+						case CharacterVerseData.StandardCharacter.Intro:
+							AddExtraBiblicalToGroupOrNarratorGroup(character.CharacterId, BookIntroductionGroup, characterType,
+								narratorPrefix, dramatizationPreferences);
+							break;
+					}
+				}
+			}
+
+			private void AddExtraBiblicalToGroupOrNarratorGroup(string characterId, CharacterGroup group,
+				CharacterVerseData.StandardCharacter characterType, string narratorPrefix, ProjectDramatizationPreferences dramatizationPreferences)
 			{
-				if (ExtraBiblicalGroup == null)
+				if (group != null)
+				{
+					group.CharacterIds.Add(characterId);
 					return;
-				ExtraBiblicalGroup.CharacterIds.AddRange(includedCharacterDetails.
-					Select(cd => cd.CharacterId).Where(c => CharacterVerseData.IsCharacterStandard(c, false) &&
-					!Groups.Any(g => g.CharacterIds.Contains(c))));
+				}
+
+				// add to a narrator group
+				var narratorGroup = GetExtraBiblicalToNarratorGroup(characterId, characterType, narratorPrefix, dramatizationPreferences);
+
+				if (narratorGroup != null)
+					narratorGroup.CharacterIds.Add(characterId);
+			}
+
+			private CharacterGroup GetExtraBiblicalToNarratorGroup(string characterId, CharacterVerseData.StandardCharacter characterType,
+				string narratorPrefix, ProjectDramatizationPreferences dramatizationPreferences)
+			{
+				var dramatizationOption = ExtraBiblicalMaterialSpeakerOption.Omitted;
+
+				switch (characterType)
+				{
+					case CharacterVerseData.StandardCharacter.BookOrChapter:
+						dramatizationOption = dramatizationPreferences.BookTitleAndChapterDramatization;
+						break;
+
+					case CharacterVerseData.StandardCharacter.ExtraBiblical:
+						dramatizationOption = dramatizationPreferences.SectionHeadDramatization;
+						break;
+
+					case CharacterVerseData.StandardCharacter.Intro:
+						dramatizationOption = dramatizationPreferences.BookIntroductionsDramatization;
+						break;
+				}
+
+				if (dramatizationOption == ExtraBiblicalMaterialSpeakerOption.Omitted)
+					return null;
+
+				var narratorId = narratorPrefix + characterId.Split(new[] { '-' }, 2)[1];
```

Comment:
I wonder if there is a clearer/faster way to write this that wouldn't involve allocating an array, creating two new strings, array element access, and then creating another new string. Since for each type of character (as determined in the above switch), we can know the exact character offset where we expect to find the dash, it *might* be worth caching and using that information to create the new string for the narrator ID. ***OR*** since the thing following the dash is always the same length, we could just use that to pick off the book ID portion.